R/zarr-array.R

```diff
@@ -1215,46 +1229,54 @@ ZarrArray <- R6::R6Class("ZarrArray",
       if(length(filters) != length(private$shape)) {
         stop("This Zarr object has ", length(private$shape), " dimensions, ", length(filters), " were supplied")
       }
-      filters <- lapply(filters, function(x) {
-        # Proceed based on type of filter
-        if(typeof(x) == "symbol") {
-          # When empty dimension, return everything
-          if(x == "") {
-            return(NULL)
-          } else {
-            stop("Unsupported filter '", as.character(x), "' supplied") 
-          }
-          
-        } else if(typeof(x) == "double") {
-          # Return single value for dimension
-          return(slice(x, x))
-        } else if(typeof(x) == "language") {
-          x <- as.list(x)
-          # Return a range (supplied via : or seq())
-          if(x[[1]] == ":") {
-            return(slice(x[[2]], x[[3]]))
-          } else if(x[[1]] == "seq") {
-            arg_names <- names(x)
-            from <- ifelse("from" %in% arg_names, x[[which("from" == arg_names)]], x[[2]])
-            to <- ifelse("to" %in% arg_names, x[[which("to" == arg_names)]], x[[3]])
-            if(length(x) > 3) {
-              stop("Slicing with step size is not supported yet")
-              by <- ifelse("by" %in% arg_names, x[[which("by" == arg_names)]], x[[4]])
-            } else {
-              by <- NA
-            }
-            return(slice(from, to, by))
-          } else if(x[[1]] == "c") {
-            stop("Custom vector slicing is not yet supported")
-            # return(eval(y))
-          } else {
-            stop("Unsupported filter '", as.character(x), "' supplied")
-          }
-        } else {
-          stop("Unsupported filter '", as.character(x), "' supplied")
-        }
-      })
-      return(self$get_item(filters))
+      # filters <- lapply(filters, function(x) {
```

Comment:
I think these lines can be removed since the code was refactored into `manage_filters`

Reply:
Done. https://github.com/Artur-man/pizzarr/commit/74686fea7430f5652d75bf4e3ca48ddb7c26f0ee